x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/180_match_operator.yml

```diff
@@ -0,0 +1,192 @@
+  - match: { error.type: verification_exception }
+  - match: { error.reason: "Found 1 problem\nline 1:19: Unknown column [something]" }
+
+---
+"match on eval column":
+  - do:
+      catch: bad_request
+      allowed_warnings_regex:
+        - "No limit defined, adding default limit of \\[.*\\]"
+      esql.query:
+        body:
+          query: 'FROM test | EVAL upper_content = to_upper(content) | WHERE upper_content MATCH "FOX" | KEEP id'
+
+  - match: { status: 400 }
+  - match: { error.type: verification_exception }
+  - match: { error.reason: "Found 1 problem\nline 1:60: MATCH requires a mapped index field, found [upper_content]" }
+
+---
+"match on overwritten column":
+  - do:
+      catch: bad_request
+      allowed_warnings_regex:
+        - "No limit defined, adding default limit of \\[.*\\]"
+      esql.query:
+        body:
+          query: 'FROM test | DROP content | EVAL content = CONCAT("ID: ", to_str(id)) | WHERE content match "fox"'
+
+  - match: { status: 400 }
+  - match: { error.type: verification_exception }
+  - match: { error.reason: "Found 1 problem\nline 1:78: MATCH requires a mapped index field, found [content]" }
+
+---
+"match after stats":
+  - do:
+      catch: bad_request
+      allowed_warnings_regex:
+        - "No limit defined, adding default limit of \\[.*\\]"
+      esql.query:
+        body:
+          query: 'FROM test | STATS count(*) | WHERE content match "fox"'
+
+  - match: { status: 400 }
+  - match: { error.type: verification_exception }
+  - match: { error.reason: "Found 1 problem\nline 1:36: Unknown column [content], did you mean [count(*)]?" }
+
+---
+"match with functions":
+  - do:
+      catch: bad_request
+      allowed_warnings_regex:
+        - "No limit defined, adding default limit of \\[.*\\]"
+      esql.query:
+        body:
+          query: 'FROM test | WHERE content MATCH "fox" OR to_upper(content) == "FOX"'
+
+  - match: { status: 400 }
+  - match: { error.type: verification_exception }
+  - match: { error.reason: "Found 1 problem\nline 1:19: Invalid condition using MATCH" }
+
+---
+"match within eval":
+  - do:
+      catch: bad_request
+      allowed_warnings_regex:
+        - "No limit defined, adding default limit of \\[.*\\]"
+      esql.query:
+        body:
+          query: 'FROM test | EVAL matches_query = content MATCH "fox"'
+
+  - match: { status: 400 }
+  - match: { error.type: verification_exception }
+  - match: { error.reason: "Found 1 problem\nline 1:34: EVAL does not support MATCH expressions" }
+
+---
+"match with non text field":
+  - do:
+      catch: bad_request
+      allowed_warnings_regex:
+        - "No limit defined, adding default limit of \\[.*\\]"
+      esql.query:
+        body:
+          query: 'FROM test | WHERE id MATCH "fox"'
+
+  - match: { status: 400 }
+  - match: { error.type: verification_exception }
+  - match: { error.reason: "Found 1 problem\nline 1:19: MATCH requires a text or keyword field, but [id] has type [integer]" }
```

Comment:
Checking my understanding - do we want `MATCH` to skip non-string fields in the future? Like, if I do `WHERE int MATCH "1"` should that fail? Or match nothing? Or match the literal `1`? It's all good to match whatever makes sense in this PR though.

Reply:
Our aim is to add full text search capabilities and not necessarily have full parity with the match query - this is why I restricted this to `text` and `keyword` for now - if we want to add support for data, boolean and numeric fields we can definitely do that later. not in this PR though

Reply:
Matching on `keyword` and `text` seems reasonable. It's not clear to me how much value there is matching on more field types, but I agree that that can be discussed separately.